### ISSUE TITLE
Not able to reterive Email ID of User.

### ISSUE THREAD

**Issue by OTBConsulting** [Wizcorp/phonegap-facebook-plugin#1160]
Hi Friends

I have successfully integrated the Facebook Plugin with our application. When I am running the same, I am not able to retrieve the Email ID from FB but I am getting the FB ID and FB Name. 

Can someone help. It is a bit urgent.

Code:

facebookConnectPlugin.api("me/?fields=id,email", ["public_profile"],
    function (result) {
        alert("Result: " + JSON.stringify(result));
        /* 
    function (error) {
        alert("Failed: " + error);
    });

**Comment by crmolinaz** [comment]
Could be a permission problems. Try this

["public_profile", "email"]

**Comment by OTBConsulting** [comment]
Hi Linaz

Thanks for your revert. We have tried including "Email" and also tried with NULL. The result is same. I am fetching the Name and FB ID but not the email.

Attaching the alert. Have been struggling since a week without a solution,.
![not getting email id](https://cloud.githubusercontent.com/assets/12799538/10726197/42c8504a-7bf7-11e5-9f14-b17158ecf142.jpg)
![erorr with public_profile _ email](https://cloud.githubusercontent.com/assets/12799538/10726198/42cef56c-7bf7-11e5-8c74-946526f70f71.jpg)


Rishit

**Comment by crmolinaz** [comment]
You need to include that permission when you try to login.

example:
facebookConnectPlugin.login(
                ['email', 'public_profile'],

**Comment by 0ruben** [comment]
Same problem here with some phones, I can't get email for some users...
Very annoying

**Comment by OTBConsulting** [comment]
Hi Ruben
Check your auth-o on FB Developer page. You need to have it on. It resolved the error for us. Rishit Sanghvi

**Comment by roysG** [comment]
Same for me, i define the settings as in the instruction and most of the time i get the user info include email but there some user that i can't get their info!

What can be the problem ?

**Comment by lightwaxx** [comment]
_below is what I did and it works
function initApp(){
    facebookConnectPlugin.login(['email', 'public_profile'],
        fbLoginSuccess,
        function (error){
            console.log(error);
        }
    );
}
function getUserInfo(){
    facebookConnectPlugin.api('me/?fields=id,name,email', ['email','public_profile'],
        function (result) {
            console.log(result);
        },
    function (error) {
        console.log(error);
    });
}

**Comment by rafaelcoutinho** [comment]
True, add the fields param to the url fixed it. Think the api is not appending the params correctly

**Comment by simerjit-fwrk** [comment]
unable to get email.. anybody solved this??
